### PR TITLE
mgr: fix python module teardown & add tests

### PR DESCRIPTION
**Pull request by jcsp**
Depends on https://github.com/ceph/ceph/pull/14220 for tests to pass.

### CODE REVIEW

**Comment by tchaikov** [comment]
will review this PR tomorrow.

File: qa/tasks/vstart_runner.py [hunk 1]
```diff
@@ -686,9 +683,6 @@ def __init__(self, ctx):
         super(LocalMgrCluster, self).__init__(ctx)
 
         self.mgr_ids = ctx.daemons.daemons['mgr'].keys()
-        if not self.mgr_ids:
-            raise RuntimeError("No manager daemonss found in ceph.conf!")
```
Comment (tchaikov) [inline qa/tasks/vstart_runner.py]: daemons.

File: qa/tasks/vstart_runner.py [hunk 2]
```diff
@@ -811,14 +805,17 @@ def scan_tests(modules):
...
     max_required_clients = 0
+    max_required_mgr = 0
 
     for suite, case in enumerate_methods(overall_suite):
         max_required_mds = max(max_required_mds,
                                getattr(case, "MDSS_REQUIRED", 0))
         max_required_clients = max(max_required_clients,
                                getattr(case, "CLIENTS_REQUIRED", 0))
+        max_required_mgr = max(max_required_mgr,
+                               getattr(case, "REQUIRE_MGRS", 0))
```
Comment (tchaikov) [inline qa/tasks/vstart_runner.py]: can we use "MGRS_REQUIRED" for the sake of consistency?

File: qa/tasks/vstart_runner.py [hunk 3]
```diff
@@ -906,7 +903,7 @@ def exec_test():
         vstart_env["FS"] = "0"
         vstart_env["MDS"] = max_required_mds.__str__()
         vstart_env["OSD"] = "1"
-        vstart_env["MGR"] = "1"
+        vstart_env["MGR"] = max(max_required_mgr, 1).__str__()
```
Comment (tchaikov) [inline qa/tasks/vstart_runner.py]: just out of curiosity, why don't use
```py
str(max(max_required_mgr, 1))
```

**Comment by tchaikov** [comment]
@jcsp lgtm modulo some nits.

**Comment by yuriw** [comment]
@jcsp pls rebase

remote: Counting objects: 21, done.
remote: Compressing objects: 100% (3/3), done.
remote: Total 21 (delta 15), reused 14 (delta 14), pack-reused 4
Unpacking objects: 100% (21/21), done.
From https://github.com/jcsp/ceph
 * branch            wip-19412  -> FETCH_HEAD
Auto-merging src/mgr/PyModules.cc
CONFLICT (content): Merge conflict in src/mgr/PyModules.cc
Auto-merging src/mgr/MgrPyModule.cc
Auto-merging qa/tasks/vstart_runner.py
Automatic merge failed; fix conflicts and then commit the result.

**Comment by liewegas** [comment]
those might be conflicts with https://github.com/ceph/ceph/pull/14507 btw

File: qa/tasks/vstart_runner.py (hunk 1 above)
Comment (jcsp) [inline qa/tasks/vstart_runner.py]: This line was being removed anyway

File: qa/tasks/vstart_runner.py (hunk 2 above)
Comment (jcsp) [inline qa/tasks/vstart_runner.py]: Done

File: qa/tasks/vstart_runner.py (hunk 3 above)
Comment (jcsp) [inline qa/tasks/vstart_runner.py]: No reason, they're completely equivalent afaik

**Comment by jcsp** [comment]
Pushed an update (this did not need rebase, as sage says it is just conflicting with the other outstanding PR)

**Comment by yuriw** [comment]
Will try to rebuild again

**Comment by yuriw** [comment]
@jcsp still see the same conflict

**Comment by jcsp** [comment]
@yuriw yes, you will see that until one or other of the PRs merges and the one one gets rebased.

**Comment by tchaikov** [comment]
@jcsp could you rebase your changes?

**Comment by jcsp** [comment]
@tchaikov done!

**Comment by tchaikov** [comment]
http://pulpito.ceph.com/kchai-2017-04-21_03:04:04-rados-wip-kefu-testing---basic-mira/

**Pull request merged**